### ISSUE TITLE
AT upload invalid theme: show better error message

### ISSUE THREAD

**Issue by lamosty** [Automattic/wp-calypso#14515]
When you have an AT (Automated Transfer) site and try to upload an invalid theme (e.g. theme having nested zips, not having style.css, etc), it shows the following error:

![selection_310](https://cloud.githubusercontent.com/assets/4988512/26498218/8b4e8a60-422e-11e7-8306-22c436f93bf6.png)

It's almost impossible to infer from that error what's wrong. For example, WP core (wp-admin) returns `The package could not be installed: The theme is missing the style.css stylesheet` or something very similar. I think we should take example of that and do the same here.

/cc @seear

**Comment by seear** [comment]
> It's almost impossible to infer from that error what's wrong

Agreed. We'll need the AT initiation API to return an appropriate error so we can show something on the upload page here.

**Comment by lamosty** [comment]
> Agreed. We'll need the AT initiation API to return an appropriate error so we can show something on the upload page here.

It's not during the transfer initiation. It's after you transfer a site and only then try to upload invalid theme. So I think the AT initiation API doesn't need to be changed since it's not called any more.

**Comment by seear** [comment]
> It's after you transfer a site and only then try to upload invalid theme

Ah, apologies.

Looks like we can probably tweak the Jetpack endpoint [here](https://github.com/Automattic/jetpack/blob/master/json-endpoints/jetpack/class.jetpack-json-api-themes-new-endpoint.php#L50) to return a more detailed message.

**Comment by JoshuaGoode** [comment]
This is still a valid issue and there are other variants such as "not a valid zip file".

![image](https://user-images.githubusercontent.com/4887830/109860505-40a6a380-7c2c-11eb-906d-cf0d2a6c9106.png)

However, this is likely very low priority.
